File: resiliences-service/Payloads/TopUpTargetSavingsRequest.cs
using System.ComponentModel.DataAnnotations;

namespace resiliences_service.Payloads
{
    public class TopUpTargetSavingsRequest
    {
        [Required]
        public long UserId { get; set; }

        [Required]
        public long WalletId { get; set; }

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Amount must be greater than 0")]
        public decimal Amount { get; set; }
    }
}

File: epay/domain/src/main/java/com/epay/domain/investment/input/TopUpTargetSavingsRequest.java
package com.epay.domain.investment.input;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class TopUpTargetSavingsRequest {

    @NotNull
    private Long userId;

    @NotNull
    private Long walletId;

    @NotNull
    @DecimalMin(value = "0.01", message = "Amount must be greater than 0")
    private BigDecimal amount;
}